src/OrchardCore.Modules/OrchardCore.Twitter/Signin/Configuration/TwitterOptionsConfiguration.cs

```diff
@@ -57,25 +60,35 @@ public void Configure(AuthenticationOptions options)
 
         public void Configure(string name, TwitterOptions options)
         {
-            if (!string.Equals(name, TwitterDefaults.AuthenticationScheme))
+            if (!String.Equals(name, TwitterDefaults.AuthenticationScheme))
             {
                 return;
             }
+
             var settings = GetSettingsAsync().GetAwaiter().GetResult();
-            options.ConsumerKey = settings.Item1?.ConsumerKey ?? string.Empty;
+            if (settings == null)
+            {
+                return;
+            }
+
+            options.ConsumerKey = settings.Item1?.ConsumerKey ?? String.Empty;
             try
             {
                 options.ConsumerSecret = _dataProtectionProvider.CreateProtector(TwitterConstants.Features.Twitter).Unprotect(settings.Item1.ConsumerSecret);
             }
             catch
             {
-                _logger.LogError("The Consumer Secret could not be decrypted. It may have been encrypted using a different key.");
+                if (_shellSettings.State == TenantState.Running)
+                {
+                    _logger.LogError("The Consumer Secret could not be decrypted. It may have been encrypted using a different key.");
```

Comment:
```suggestion
                    _logger.LogError("The Twitter Consumer Secret could not be decrypted. It may have been encrypted using a different key.");
```

Reply:
Good catch!